configuration/src/main/java/com/synopsys/integration/configuration/help/PropertyInfo.java

```diff
@@ -0,0 +1,27 @@
+package com.synopsys.integration.configuration.help;
+
+import com.synopsys.integration.configuration.property.Property;
+
+public class PropertyInfo {
```

Comment:
I like the idea of a Property Info class but I don't think in this case it is neccessary. All we really want is a map of key to masked value.

Reply:
PropertyConfigurationHelpContext needs the actual property in addition to the key/value.  I also figure other potential consumers might want all three values

Reply:
I'm mostly thinking in terms of consistency for PropertConfiguration because in any of the other returns we don't return the Property. To me this seems like an extension to getRaw, it could be getRawMasked() and we just return a Map<String, String> where the value is the masked raw. The consumer can then themselves iterate the known properties and do !maskedMap.containsKey(property.getKet()) {continue}.

Reply:
Does it make a different that the collection of PropertyInfo is in a different class than PropertyConfiguration now?

Reply:
Yeah, I don't mind so much if it doesn't live in PropertyConfig. Only other thing is then I would move it out of the configuration package because it's a "detect" special and ideally the Configuration package is 'pure'.

Reply:
But I don't love the names. PropertyInfo is vague. PropertInfoCollector is vague. MaskedPropertyValue and MaskedPropertyValueCollector would be preferred imo (and I think I'd even push for not using Collector because that means something in the Java world, ie Collectors.joining).

Reply:
Hmm how about PropertyKeyValue and PropertyKeyValueFinder?  Also I disagree about querying the keys and values for a list of specified properties being a "Detect special".   I think other configuration consumers could be interested in that information/it would make sense for the utility to exist within the module

Reply:
But these are not 'property values'. If you were a 'blind' consumer and you called these methods 'getValues' or 'getInfo' thinking you were getting actual values, and tried to use the values, that would be incorrect. First you'd only be getting the raw 'value' and second your value might be masked. There is a dramatic difference between a 'value' and the 'raw string' and the 'masked raw string'. For example 'BooleanProperty' a valid 'value' for it is not the string "bool" as you would get here, it is an actual boolean value. The 'raw value' might be the string but it could also be anything the user passed. And even still the 'masked raw value' might be "****" and if consumer code tried to use it, they wouldn't work. 

It is fair that other consumers would be interested in these values for printing, that is a good point. I think having a 'getRaw(Set<Property>)' would make sense. 

I think then a PropertyMasker could make sense. Say he takes in a Set<Property> known, Map<String, String> raw, Predicate<bool> shouldMask and returns a set of masked raw Map<String, String> or even Map<Property, String>.